### DISCUSSION TITLE
Update terminology: Use "witness" instead of "proof"

### DISCUSSION THREAD

**Discussion by matheus23** [ucan-wg/spec#23] [Ideas]
Something @expede and I have been talking about.
We think you'd usually call what today is contained in a UCAN's "proofs" section "witness".

From my intuition (and I'm not sure about it!) a proof is something that can be "right or wrong" on its own. However, witnesses are just witnesses. They can't be right or wrong on their own, but they can be *used* to prove something.

**Comment by expede** [comment]
Kind of confusingly, "witnesses" are used for "proof", so we can say that the `prf` section is "proof by these witnesses".

**Comment by expede** [comment]
As an intuition:

Alice: "You say unicorns are real? What's your proof?" [request for proof]
Bob: "Here's a unicorn 🦄" [witness]

**Reply by Gozala** [reply]
I think this is inaccurate if Bob shows unicorn than unicorn is a proof if Bob says he's seen one than Bob is a witness.

**Comment by bgins** [comment]
> Bob: "Here's a unicorn 🦄" [witness]

Is "witness" a noun or verb here? Is Bob a witness who can provide proof or is the act of providing proof the witness?

**Comment by expede** [comment]
@bgins Great question!

The witness is the thing that proves the rule, in this case the specific unicorn 🦄 that Bob is pointing at.

A unicorn is a witness to the fact that at least one unicorn exists. `2` is a witness to the existence of even numbers. This comes from constructive mathematics, where proof is done by showing that you can construct an object that matches your ~~predicate~~ statement.

https://en.wikipedia.org/wiki/Witness_(mathematics)

**Reply by bgins** [reply]
> A unicorn is a witness to the fact that at least one unicorn exists. 2 is a witness to the existence of even numbers.

I see, the witness is the evidence that something exists. Witness is slightly confusing because in other areas like the law, a witness is someone who provides evidence, but not the evidence itself. Or to witness is to observe some evidence.

**Reply by expede** [reply]
Yeah, totally. In colloquial terms you can witness a robbery, and the CCTV footage is evidence. This terminology has a different root. This is part of why we avoided the term "witness" previously, but I've been thinking about tightening up the technical terms — maybe going too deep 🤔

**Reply by bgins** [reply]
Maybe "evidence" is worth considering? Colloquially, evidence isn't proof on its own. On the other hand, it's less of a technical term.

**Comment by expede** [comment]
The other thing to note in this system, which is relevant to UCAN, is that (by default) you loose the law of the excluded middle. Things are not "true or false". They are "true" if you can provide a witness, or "I dunno, maybe" otherwise. In the same way that I can't prove that there isn't a unicorn in the wild — there _might_ be 🤷‍♀️ 

(As an aside, when as a child my partner learned that unicorns weren't real, she also applied this to narwhals. She was many years later shocked to discover that sea unicorns are in fact real 😜)

For UCAN, I can't prove that a particular DID _can't_ do something unless it contradicts the capability's semantics (e.g. can't `SEND` a DNS record). You always _might_ have a capability hidden somewhere that's not shared with me that can act as a witness to a capability.

**Reply by bgins** [reply]
> (As an aside, when as a child my partner learned that unicorns weren't real, she also applied this to narwhals. She was many years later shocked to discover that sea unicorns are in fact real 😜)

That's great. And yeah, we can't be sure. It just takes one 🦄 witness and they're real. 😄 

> For UCAN, I can't prove that a particular DID can't do something unless it contradicts the capability's semantics (e.g. can't SEND a DNS record). You always might have a capability hidden somewhere that's not shared with me that can act as a witness to a capability.

That makes sense. All authority might exist. We only know what we are talking about here and now.

**Comment by expede** [comment]
@matheus23 I converted this into a discussion for now. If the conversation moves back to changing the field label, we can flip it back!

**Reply by Gozala** [reply]
I personally don't like "witness" because has several meaning and first suggested by Google (which also matches what comes to my mind is)

> a person who sees an event, typically a crime or accident, take place.

And to me witness is not a right analogy here, because it implies trusting a "witness". For this reason I think proof is much better term, I have access here is the proof 🎫. 

If name needs to change I think evidence would be a much better option

> the available body of facts or information indicating whether a belief or proposition is true or valid

**Reply by expede** [reply]
Fair! We're using it in the mathematical and cryptographic sense. That's very unfamiliar to most people, even if "more accurate." I'm fine with leaving it as "proof", since that's both close enough in meaning and seems clear to most people so far
